src/python_framework_v02/troute/nhd_io.py

```diff
@@ -469,7 +469,98 @@ def get_channel_restart_from_wrf_hydro(
 
     return q_initial_states
 
+def write_channel_restart_to_wrf_hydro(
+    data,
+    restart_files,
+    channel_initial_states_file,
+    dt_troute,
+    nts_troute,
+    crosswalk_file,
+    channel_ID_column,
+    restart_file_dimension_var = 'links',
+    troute_us_flow_var_name = 'qlink1_troute',
+    troute_ds_flow_var_name = 'qlink2_troute',
+    troute_depth_var_name = 'hlink_troute',
+):
+    """
+    Write t-route flow and depth data to WRF-Hydro restart files. New WRF-Hydro restart
+    files are created that contain all of the data in the original files, plus t-route
+    flow and depth data. 
+    
+    Agruments
+    ---------
+        data (Data Frame): t-route simulated flow, velocity and depth data
+        restart_files (list): globbed list of WRF-Hydro restart files
+        channel_initial_states_file (str): WRF-HYDRO standard restart file used to initiate t-route simulation
+        dt_troute (int): timestep of t-route simulation (seconds)
+        nts_troute (int): number of t-route simulation timesteps
+        crosswalk_file (str): File containing reservoir IDs IN THE ORDER of the Restart File
+        channel_ID_column (str): field in the crosswalk file to assign as the index of the restart values
+        restart_file_dimension_var (str): name of flow and depth data dimension in the Restart File
+        troute_us_flow_var_name (str):
+        troute_ds_flow_var_name (str):
+        troute_depth_var_name (str):
+        
+    Returns
+    -------
+    """
+    # create t-route simulation timestamp array
+    with xr.open_dataset(channel_initial_states_file) as ds:
+        t0 = ds.Restart_Time.replace('_', ' ')
+    t0 = np.array(t0,dtype = np.datetime64)
+    troute_dt = np.timedelta64(dt_troute, 's')
+    troute_timestamps = t0 + np.arange(nts_troute) * troute_dt
```

Comment:
Un until now, we have been working in the timestep space. That is, each column of the flowveldepth dataframe is indexed by a timestep, not a timestamp. In order to put the correct data into restart files, we need to work in the timestamp space.